### PR TITLE
IJMP-1660-Add-new-TSO-request

### PR DESCRIPTION
**Pull request by ATsikhamirau**
1) added new tso request for z/os versions > 2.3

### CODE REVIEW

**Review by dnlsv — APPROVED** [review]

File: gradle.properties [hunk 1]
```diff
@@ -24,7 +24,7 @@ group=org.zowe
 archivesName=kotlinsdk
 
 # Artifacts version
-version=0.5.0-rc.7
+version=0.5.0-rc.9
```
Comment (KUGDev) [inline gradle.properties]: no need to manually update

File: src/main/kotlin/org/zowe/kotlinsdk/TsoApi.kt [hunk 2]
```diff
@@ -101,4 +101,19 @@ interface TsoApi {
...
+  /**
+   * An API function to send a command for execution in the TSO address space
+   * @param authorizationToken is a base 64 encoding representation of *userid*:*password*
+   * @param contentType content type of the request
+   * @param body wrapped instance of TsoRequestBody class
+   * @return a wrapped instance of [TsoResponseEnhanced]
+   */
+  @AvailableSince(ZVersion.ZOS_2_4)
+  @PUT("/zosmf/tsoApp/v1/tso")
+  fun executeCommandInTsoAddressSpace(
```
Comment (KUGDev) [inline src/main/kotlin/org/zowe/kotlinsdk/TsoApi.kt]: executeTsoCommand

File: src/main/kotlin/org/zowe/kotlinsdk/TsoResponseEnhanced.kt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+ *
+ * Copyright IBA Group 2020
+ */
+
+package org.zowe.kotlinsdk
+
+import com.google.gson.annotations.Expose
+import com.google.gson.annotations.SerializedName
+
+data class TsoResponseEnhanced(
```
Comment (KUGDev) [inline src/main/kotlin/org/zowe/kotlinsdk/TsoResponseEnhanced.kt]: TsoCmdResponse

File: src/main/kotlin/org/zowe/kotlinsdk/TsoResponseEnhanced.kt [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    @SerializedName("tsoPromptReceived")
+    @Expose
+    var tsoPromptReceived: String? = null,
+
+    @SerializedName("keywordDetected")
+    @Expose
+    var keywordDetected: String? = null
+)
+
+data class TsoResult(
```
Comment (KUGDev) [inline src/main/kotlin/org/zowe/kotlinsdk/TsoResponseEnhanced.kt]: TsoCmdResult

File: src/main/kotlin/org/zowe/kotlinsdk/TsoResponseEnhanced.kt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    @SerializedName("returnCode")
+    @Expose
+    var returnCode: String? = null,
+
+    @SerializedName("reason")
+    @Expose
+    var reason: String? = null
+)
+
+data class TsoRequestBody(
```
Comment (KUGDev) [inline src/main/kotlin/org/zowe/kotlinsdk/TsoResponseEnhanced.kt]: TsoCmdRequestBody

File: src/main/kotlin/org/zowe/kotlinsdk/TsoResponseEnhanced.kt [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    @SerializedName("servletKey")
+    @Expose
+    var servletKey: String? = null,
+
+    @SerializedName("keyword")
+    @Expose
+    var keyword: String? = null
+)
+
+enum class CmdState(private val cmdState: String) {
```
Comment (KUGDev) [inline src/main/kotlin/org/zowe/kotlinsdk/TsoResponseEnhanced.kt]: TsoCmdState

**Review by KUGDev — CHANGES_REQUESTED** [review]

**Review by KUGDev — APPROVED** [review]

**Pull request merged**